pkg/overlay/config_test.go

```diff
@@ -4,35 +4,99 @@ package overlay
 
 import (
 	"context"
+	"fmt"
+	"math/rand"
+	"net"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
 
+	"storj.io/storj/pkg/auth"
 	"storj.io/storj/pkg/kademlia"
+	"storj.io/storj/pkg/provider"
+	"storj.io/storj/pkg/statdb"
+	statpb "storj.io/storj/pkg/statdb/proto"
 )
 
 func TestRun(t *testing.T) {
+	bctx := context.Background()
+	ctxWithAPIKey := auth.WithAPIKey(bctx, []byte(""))
+
 	kad := &kademlia.Kademlia{}
 	var kadKey kademlia.CtxKey
-	ctxWithKad := context.WithValue(context.Background(), kadKey, kad)
+	ctxWithKad := context.WithValue(ctxWithAPIKey, kadKey, kad)
+
+	prv, address, err := getProvider(ctxWithKad)
```

Comment:
Use testplanet to get a provider, it's faster and should be less code.

Reply:
I get the following error when I try to use testplanet here:
```
# storj.io/storj/pkg/overlay
import cycle not allowed in test
package storj.io/storj/pkg/overlay (test)
    imports storj.io/storj/internal/testplanet
    imports storj.io/storj/pkg/overlay
```
Is there a way to get around this? Should I attempt to get around this? The only reason I need the provider is so that `prv.Identity()` can be use for the statdb client identity.